pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the QSVT template and qsvt wrapper function.
+"""
+import numpy as np
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.ops import BlockEncode, PCPhase
+from pennylane.ops.op_math import adjoint
+from pennylane.operation import AnyWires, Operation
+
+
+def qsvt(A, phi_vect, wires):
+    """Performs the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit using
+    :class:`~.BlockEncode` and :class:`~.PCPhase`.
+
+    Args:
+        A (Array): the general :math:`(n \times m)` matrix to be encoded.
+        phi_vect (list[Float]): a list of angles by which to shift to obtain the desired polynomial.
+        wires (Iterable): the wires the template acts on.
+
+    .. details::
+        :title: Usage Details
+
+        .. code-block:: python
+
+            dev = qml.device('default.qubit', wires=2)
+            A = [[0.1, 0.2], [0.3, 0.4]]
+            lst_phis = [0.1, 0.2, 0.3]
+            @qml.qnode(dev)
+            def example_circuit(A):
+                qml.qsvt(A, lst_phis, wires=[0,1])
+                return qml.expval(qml.PauliZ(wires=0))
+
+            qml.draw(example_circuit)(A)
+    """
+    c, r = qml.math.shape(A)
+
+    U_A = qml.BlockEncode(A, wires=wires, do_queue=False)
+    lst_projectors = []
+    for idx, phi in enumerate(phi_vect):
+        if idx % 2 == 0:
+            lst_projectors.append(PCPhase(phi, dim=r, wires=wires, do_queue=False))
+        else:
+            lst_projectors.append(PCPhase(phi, dim=c, wires=wires, do_queue=False))
+
+    return QSVT(U_A, lst_projectors, wires=wires)
+
+
+class QSVT(Operation):
```

Comment:
If this is the template, then it must appear on the template page. Right now, I don't see that happening in the docs.